components/alert/AlertMap.vue

```diff
@@ -711,250 +867,181 @@ export default Vue.extend({
           }
-        });
-        
-        // TODO: This should probably go on the region instead
-        region.outlineStamp = this.$L.stamp(polygon);
-        this.polys.addLayer(polygon);
-      });
+        })
+
+        region.outlineStamp = this.$L.stamp(polygon)
+        this.polys.addLayer(polygon)
+      })
 
       // Build facility links
       regions.forEach((region) => {
         region.connections.forEach((connection) => {
-          var link = this.$L.polyline([
-            worldToMap(region.badgeLocation.asArray()), 
-            worldToMap(connection.badgeLocation.asArray())
-            ], {
+          // No duplicates
+          if (region.linkStamps.get(connection.id) !== undefined) {
+            return
+          }
+          const link = this.$L.polyline(
+            [
+              worldToMap(region.badgeLocation.asArray()),
+              worldToMap(connection.badgeLocation.asArray()),
+            ],
+            {
               weight: 2,
               color: '#FFFFFF',
               opacity: 0.6,
-              pane: "linkPane",
+              pane: 'linkPane',
               interactive: false,
-              bubblingMouseEvents: true
-          });
-          // TODO: This should probably go on region and connection
-          this.linkStamps[region.id.toString() + " " + connection.id.toString()] = this.$L.stamp(link);
-          
-          var bglink = this.$L.polyline([
-            worldToMap(region.badgeLocation.asArray()), 
-            worldToMap(connection.badgeLocation.asArray())
-            ], {
+              bubblingMouseEvents: true,
+            }
+          )
+          region.linkStamps.set(connection.id, this.$L.stamp(link))
+          connection.linkStamps.set(region.id, this.$L.stamp(link))
+
+          const bglink = this.$L.polyline(
+            [
+              worldToMap(region.badgeLocation.asArray()),
+              worldToMap(connection.badgeLocation.asArray()),
+            ],
+            {
               weight: 2,
               color: MAP_LINK_COLORS[5]?.toString(),
               opacity: 0.0,
-              pane: "linkPane",
-          });
+              pane: 'linkPane',
+            }
+          )
           // TODO: This should probably go on region and connection
-          this.linkStamps["bg" + region.id.toString() + " " + connection.id.toString()] = this.$L.stamp(bglink);
+          region.bgLinkStamps.set(connection.id, this.$L.stamp(bglink))
+          connection.bgLinkStamps.set(region.id, this.$L.stamp(bglink))
 
-          this.links.addLayer(bglink);
-          this.links.addLayer(link);
-        });
-        this.createRegionBadge(region);
-      });
+          this.links.addLayer(bglink)
+          this.links.addLayer(link)
+        })
+        this.createRegionBadge(region)
+      })
 
-      this.map.on("zoom", (ev: L.LeafletEvent) => {
-        for(var region of this.mapRegions.values()) {
+      this.map.on('zoom', () => {
```

Comment:
Wouldn't this be where the sound is emitted?

Reply:
Yep, forgot I had a zoom event handler for the map already